src/support.js

```diff
@@ -282,7 +282,7 @@ function addFunction(func) {
 
 function removeFunction(index) {
 #if EMULATED_FUNCTION_POINTERS == 0
-  functionPointers[(index-{{{ FUNCTION_POINTER_ALIGNMENT }}})/{{{ FUNCTION_POINTER_ALIGNMENT }}}] = null;
+  functionPointers[index-1] = null;
```

Comment:
hmm, this line should just be removed, I think - it was only valid when `FUNCTION_POINTER_ALIGNMENT` was 2. When it's 1 it will null out something in a dangerous way, unless I'm missing something.

Reply:
(it wasn't even valid before, if I understand it correctly - it just happened to not cause any harm...)

Reply:
Why? `addFunction` fills up the `functionPointers` array from 0 but it returns the actual index + 1, and when the given index is 1, `removeFunction` would just delete the function added in `functionPointers[0]`. I think the reason `addFunction` returns `index + 1` is in asm.js every function table (each with different signature) has a null function at its index 0.

Reply:
Yes, I think you're right.

But that means that a few lines down, the `tables[sig][index] = null;` is wrong, it should have been `index - 1` all along?

Reply:
Oh, I guess not, the indexing is just different there.